### ISSUE TITLE
Valid flush at end of script is blocking

### ISSUE THREAD

**Issue by dpwspoon** [k3po/k3po#128]
The flush isn't useful here, but it blocks the robot execution

{code}
accept tcp://localhost:8080
accepted
connected
read "GET"

write "HTTP/1.1 505 HTTP Version Not Supported" "\r\n"
write "\r\n"
write flush

connect tcp://localhost:8080
connected
write "GET / " ${http:randomInvalidVersion()} "\r\n"
write "Host: Robot\r\n"
write "\r\n"

read "HTTP/1.1 505"

{code}

**Comment by cmebarrow** [comment]
The issue here is write flush for tcp transport actually does nothing and does not fire the future. Details: we call Channels.flush() which sends a DownstreamFlushEvent  on the channel pipeline. This ends up in eventSunk method in NioServerSocketPipelineSink (or NioClientSocketPipelineSink if it's a client connection) which does nothing at all with the event, doesn't even fulfil it's future!

**Comment by dpwspoon** [comment]
I think there should never be a use for flush at tcp level in the scripting language, so its probably better to make it impossible to script it this way.  Rather then worrying about tcp transport

**Comment by dpwspoon** [comment]
I take back what I said above.  According this story tcp should buffer until explicit or implicit flushes: https://github.com/k3po/k3po/issues/39

**Comment by jfallows** [comment]
Agreed. Simple fix here is to mark the flush future as `success` on TCP for now since it does no work and is therefore trivially correct.
